tests/launch.rs

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg(feature = "openssl")]
+
+use sev::certs::Chain;
+use sev::firmware::Firmware;
+use sev::launch::{HeaderFlags, Launcher, Policy};
+use sev::session::Session;
+
+use sev::Generation;
+
+use kvm_bindings::kvm_userspace_memory_region;
+use kvm_ioctls::{Kvm, VcpuExit};
+use mmarinus::{perms, Kind, Map};
+
+use std::convert::TryFrom;
+use std::fs::File;
+use std::io::Write;
+
+const CODE: &[u8] = &[
+    0xf4, // hlt
+];
+
+fn get_cert_chain(sev: &mut Firmware) -> Chain {
+    use codicon::Decoder;
+
+    // If you are running this test suite often without manipulating the platform
+    // state (i.e., platform reset) it is recommended that you export the entire
+    // chain to a file and use this environment variable to avoid getting rate-limited
+    // by the AMD web service (which will cause a panic).
+    if let Ok(cached_chain) = std::env::var("CHAIN") {
+        let mut chain = File::open(cached_chain).unwrap();
+
+        Chain::decode(&mut chain, ()).unwrap()
+    } else {
+        let mut platform = sev.pdh_cert_export().unwrap();
+
+        let id = sev.get_identifer().unwrap();
+        let url = format!("https://kdsintf.amd.com/cek/id/{}", id);
+
+        let mut resp = reqwest::blocking::get(&url)
+            .unwrap()
+            .error_for_status()
+            .unwrap();
+
+        let mut cek = vec![];
+        let _ = resp.copy_to(&mut cek).unwrap();
+        platform.cek = sev::certs::sev::Certificate::decode(&mut &cek[..], ()).unwrap();
+
+        let ca = Generation::try_from(&platform).unwrap().into();
+
+        Chain { sev: platform, ca }
+    }
+}
+
+#[cfg_attr(not(has_sev), ignore)]
+#[test]
+fn sev() {
+    let mut sev = Firmware::open().unwrap();
+    let build = sev.platform_status().unwrap().build;
+    let chain = get_cert_chain(&mut sev);
+
+    let policy = Policy::default();
+    let session = Session::try_from(policy).unwrap();
+    let start = session.start(chain).unwrap();
+
+    let kvm = Kvm::new().unwrap();
+    let vm = kvm.create_vm().unwrap();
+
+    const MEM_SIZE: usize = 0x1000;
+    let mut address_space = Map::map(MEM_SIZE)
+        .anywhere()
+        .anonymously()
+        .known::<perms::ReadWrite>(Kind::Private)
+        .unwrap();
+
+    let mem_region = kvm_userspace_memory_region {
+        slot: 0,
+        guest_phys_addr: 0,
+        memory_size: MEM_SIZE as _,
```

Comment:
```suggestion
        memory_size: address_space.size() as _,
```

This makes it more obvious.